lib/msgpack/timestamp.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module MessagePack
+  Timestamp = Struct.new(:sec, :nsec)
+
+  class Timestamp # a.k.a. "TimeSpec"
+    TYPE = -1 # timestamp extension type
+
+    TIMESTAMP32_MAX_SEC = 0x100000000 # 32-bit unsigned int
+    TIMESTAMP64_MAX_SEC = 0x400000000 # 34-bit unsigned int
+
+    def self.from_msgpack_ext(data)
+      case data.length
+      when 4
+        # timestamp32 = (uint32be)
+        sec = data.unpack('L>').first
+        new(sec, 0)
+      when 8
+        # timestamp64 = (uint32be, uint32be)
```

Comment:
This comment is misleading... actually it's `(uint30be, uint34be)`.
I understood that `(uint32be, uint32be)` is the format to unpack the binary, but it confused me for a while.